pkg/project/resource_project.go

```diff
@@ -217,9 +217,20 @@ func projectResource() *schema.Resource {
 		},
 
 		"repos": {
-			Type:        schema.TypeSet,
-			Optional:    true,
-			Elem:        &schema.Schema{Type: schema.TypeString},
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			MinItems: 0,
+			MaxItems: func() int {
+				defaultMaxItems := "100"
+				value, err := strconv.Atoi(getEnvVar("REPO_LIMIT_OVERRIDE", defaultMaxItems))
+				if err != nil {
+					return func() int { r, _ := strconv.Atoi(defaultMaxItems); return r }()
+				}
+				return value
```

Comment:
Why not just do the `Atoi()` once here at the very end?

Reply:
new method `getEnvIntVar` to reduce the number of calls to `strconv.Atoi`